internal/sidekick/internal/parser/discovery/message_fields.go

```diff
@@ -47,38 +46,33 @@ func makeField(model *api.API, messageID string, input *property) (*api.Field, e
 		// TODO(#2265) - handle inline object...
 		return nil, nil
 	}
-	return makeScalarField(model, messageID, input)
+	return makeScalarField(model, message, input.Name, input.Schema)
 }
 
-func makeArrayField(model *api.API, messageID string, input *property) (*api.Field, error) {
+func makeArrayField(model *api.API, message *api.Message, input *property) (*api.Field, error) {
 	if input.Schema.ItemSchema.Type == "object" && input.Schema.ItemSchema.Properties != nil {
 		// TODO(#2265) - handle inline object...
 		return nil, nil
 	}
-	typez, typezID, err := scalarType(model, messageID, input.Name, input.Schema.ItemSchema)
+	field, err := makeScalarField(model, message, input.Name, input.Schema.ItemSchema)
 	if err != nil {
 		return nil, err
 	}
-	return &api.Field{
-		Name:          input.Name,
-		JSONName:      input.Name, // OpenAPI field names are always camelCase
-		Documentation: input.Schema.Description,
-		Typez:         typez,
-		TypezID:       typezID,
-		Repeated:      true,
-		// TODO(#2268) - deprecated fields?
-	}, nil
+	field.Documentation = input.Schema.Description
+	field.Repeated = true
+	field.Optional = false
```

Comment:
style comment: I think it is slightly cleaner as it was vs. using `makeScalarField` and having to override values.

No changes needed, just giving an opinion.